### ISSUE TITLE
Test SSP IMEX with limiter, with the most stable configuration (2 or 3 Newton iterations), using `:zalesak` vertical upwinding ON for tracers, energy and density.

### ISSUE THREAD

**Issue by valeriabarra** [CliMA/ClimaAtmos.jl#1531]
Part of #1310

**Comment by valeriabarra** [comment]
Same comment as in the other Issue (please see this [comment](https://github.com/CliMA/ClimaAtmos.jl/issues/1513#issuecomment-1496650430)).

**Comment by valeriabarra** [comment]
For the regular CI set up, the moist HS high top test case seems to stall in different builds after 1h and 11m runtime (see [build 8750](https://buildkite.com/clima/climaatmos-ci/builds/8750#01874da5-6dfe-42ed-a8c0-eb81fc75a86f) and [build 8751](https://buildkite.com/clima/climaatmos-ci/builds/8751#01874de9-3205-471a-981a-0b4340ab1163) and we see "Exited with status -1 (agent lost)", for `dt 100secs`. 
When tried with a smaller dt, `dt 75secs`, it crashed (see [build 8756](https://buildkite.com/clima/climaatmos-ci/builds/8756#01874e3d-58e2-4341-afdf-1a3abbf18234)). 

I am investigating this locally.

**Comment by valeriabarra** [comment]
This configuration is unstable also in the regular (coarse) pipeline set up.

This is the longest I could get the moist baroclinic wave to run: 5 days, with the following CLI options:
```
julia --project=examples examples/hybrid/driver.jl --initial_condition MoistBaroclinicWave --moist equil --precip_model 0M --job_id sphere_zalesak_all_upwind_ssp_baroclinic_wave_rhoe_equilmoist_dt150_newton3 --regression_test true --dt 150secs --dt_save_to_disk 1days --ode_algo SSP333 --max_newton_iters 3 --tracer_upwinding zalesak --energy_upwinding zalesak --density_upwinding zalesak
```
At day 5, right before the crash, this is how the qt looks like, at different vertical levels (10 level in total):

- `qt`, day 5, level 7:
![rho_q_tot_day5_level7](https://user-images.githubusercontent.com/39932030/230257059-e79cd1ac-8035-460f-a5cd-15ff56d8c1e7.png)

- `qt`, day 5, level 8:
![rho_q_tot_day5_level8](https://user-images.githubusercontent.com/39932030/230257079-0f73391c-fe46-4cba-bbf7-2f732a2e655b.png)

- `qt`, day 5, level 9:
![rho_q_tot_day5_level9](https://user-images.githubusercontent.com/39932030/230257113-8a55ff8b-7079-49c4-b812-b89682cdfa2b.png)

- `qt`, day 5, level 10:
![rho_q_tot_day5_level10](https://user-images.githubusercontent.com/39932030/230257144-1697807b-cb9e-4057-bd5a-fb20d4960318.png)


And here are the extrema for `qt`:
```
println(ρq_tot |> parent |> extrema)
(-1.8088796f-8, 0.018971413f0)
```

I tried several time steps and Newton iterations but nothing helped.

I also tried running explicit only (to rule out any potential problem with the approximated Jacobian implementation), but it didn't help either.

I think we can close this issue and investigate this separately.
